runtime/jvmti/jvmtiThread.c

```diff
@@ -1300,18 +1317,160 @@ jvmtiSuspendAllVirtualThreads(jvmtiEnv *env,
 	jint except_count,
 	const jthread *except_list)
 {
-	assert(!"jvmtiSuspendAllVirtualThreads unimplemented");
-	return JVMTI_ERROR_NONE;
-}
+	J9JavaVM *vm = JAVAVM_FROM_ENV(env);
+	J9VMThread *currentThread = NULL;
+	jvmtiError rc = JVMTI_ERROR_NONE;
+
+	Trc_JVMTI_jvmtiSuspendAllVirtualThreads_Entry(env);
+
+	rc = getCurrentVMThread(vm, &currentThread);
+	if (JVMTI_ERROR_NONE == rc) {
+		jint i = 0;
+		BOOLEAN currentThreadEverSuspended = FALSE;
+		J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+
+		vmFuncs->internalEnterVMFromJNI(currentThread);
+
+		ENSURE_PHASE_LIVE(env);
+		ENSURE_CAPABILITY(env, can_suspend);
+		ENSURE_CAPABILITY(env, can_support_virtual_threads);
 
+		ENSURE_NON_NEGATIVE(except_count);
+		if ((NULL == except_list) && (0 != except_count)) {
+			JVMTI_ERROR(JVMTI_ERROR_NULL_POINTER);
+		}
+
+		for (i = 0; i < except_count; ++i) {
+			jthread thread = except_list[i];
+			if ((NULL == thread)
```

Comment:
Strictly speaking, this check is incomplete - it's legal to have a JNI ref to NULL (think weak global ref). There's manu other occurrences of this error around, so we can address this in a future PR.

Reply:
Nothing yet, so please open.

Reply:
Opened: https://github.com/eclipse-openj9/openj9/issues/16081.